modules/ROOT/pages/http/http-listener-ref.adoc

```diff
@@ -1,182 +1,296 @@
+By default, a regular response will return a 200 status code and the message payload as body, while an error response will return a 500 status code with the flow error's description as body. These responses can be customized by providing:
 
-Besides headers and a body, an HTTP request is also composed of a request line. The HTTP request line is the content on the first line of the HTTP Request that includes the URI from the address bar of the client request. For example:
+* the status code
+* the reason phrase
+* the headers multi map
+* the body
 
-`POST /mydomain/login/?user=aaron&age=32 HTTP/1.1`  +
+DataWeave can be used to generate each one and variables can be used to propagate data from the flow.
 
-This content is transformed into the following attributes in the Mule Message:
+In the following example, an endpoint is defined to store logos for an account.
 
-Property names are defined as follows:
+When the storage is successful, a `201 Created` is returned with a body such as `Corporate Logo has been stored as a mulesoft logo.`.
 
-* `#[attributes.method]`
-+
-The HTTP Request method
-+
-* `#[attributes.listenerPath]`
-+
-The path: `/mydomain/login/`
-+
-* `#[attributes.queryParams.user]` and `#[attributes.queryParams.age]`
-+
-The query string values from the URI, `aaron` and `32`.
-+
-* `#[attributes.queryString]`
-+
-The query string from the URI, without parsing, `user=aaron&age=32`.
+When the storage fails, the status code is defined through the `errorCode` variable if available or `500` by default. Additionally, a custom header is added (`X-Time`) and a body such as `Corporate Logo could not be stored.` is returned.
+Should there be a `CONNECTIVITY` error storing the logo, for example, the returned status code would be `504` while any other errors would result in a `500`.
 
-When the HTTP Listener has URI parameter placeholders in its path, a Map is created with the holder name as key and the value as extracted from the request path. For example, `domain=mydomain`. The Listener path must be defined as `{domain}/login`
+It's important to note that while a reason phrase is not defined for error responses,
+ the connector will attempt to define it based on the status code. Thus, a `Gateway Timeout` or `Internal Server Error` will be returned in the scenarios explained before.
 
-== Generating the HTTP Response
-
-After request processing, the message is returned to the HTTP connector to provide a response. You can customize the response body, attachment, headers, and status.
-
-=== HTTP Response Body
-
-The HTTP Response body is generated from payload of the Mule Message after executing the flow unless explicitly configured. You can use a DataWeave script or expression to customise it.
-
-=== HTTP Response Headers
+[source,xml,linenums]
+----
+<flow name="store-logo">
```

Comment:
for this kind of large flows, you can use numeric callouts to add line by line explanations below. Check the SDK docs, it's full of examples of this and we found it really helps readability.